Bottom: e095de5d3ae86aa69b264ca781aa956a9adc1d42
Top:    c03bd1ac907421a7f693fbeb2bb277dab6c9c707
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-05 18:49:24 +0800

autofs-5.0.7 - fix memory leak in get_dc_list()

In get_dc_list() if an invalid port is found the allocated list storage
isn't freed on the error exit.


---

diff --git a/modules/dclist.c b/modules/dclist.c
index d16b913..bd4d5e6 100644
--- a/modules/dclist.c
+++ b/modules/dclist.c
@@ -549,6 +549,8 @@ struct dclist *get_dc_list(unsigned int logopt, const char *uri)
 					error(logopt,
 					      "invalid port: %u", dcs[i].port);
 					free_srv_rrs(dcs, numdcs);
+					/* Ensure realloced list is freed */
+					list = tmp;
 					goto out_error;
 				}
 				strcat(tmp, port);
